### PR TITLE
i.landsat.acca: add test cases 

### PR DESCRIPTION
**Pull request by codersmiti**
This PR introduces a test suite for the i.landsat.acca GRASS GIS module, covering a range of scenarios, including cloud mask creation, cloud shadow detection, and the effect of different input parameters on cloud identification. The test cases ensure correct functionality across scenarios like thermal correlation, binary output validation, and the impact of the b45ratio threshold.

**Key updates in this PR include:**

- **Functionality:** Validate cloud mask creation using i.landsat.acca and ensure proper cloud detection.

- **Edge Cases**: Test cloud shadow detection and the impact of the b45ratio parameter on cloud detection.

- **Cloud Mask Validations**: Ensure the cloud mask is binary (with only two unique values) and accurately represents cloud and non-cloud pixels, while validating that cloud pixels have higher thermal values than non-cloud pixels.

**Test Case Additions:**

-  **Basic Functionality:**
      - Verify creation and existence of the cloud mask output.  
      - Test cloud mask creation for different input data.

- **Edge Cases:**

     -  Cloud Shadow Detection: Test cloud shadow detection by ensure that shadow pixels are 
                                                  detected, and the output is valid.

     - b45ratio Parameter Effect: Test the impact of the b45ratio parameter on cloud detection by adjusting it. Ensure that a higher b45ratio threshold results in fewer detected cloud pixels, indicating a more conservative approach to cloud identification.

-  **Cloud and Thermal Correlation Validation:**
   -  Binary Cloud Mask: Verify that the cloud mask output is binary, containing only two unique values (typically 0 for non-clouds and 1 for clouds), ensuring correct classification.
   -  Thermal Band Correlation: Validate the correlation between detected cloud pixels and thermal band values. Ensure that cloud pixels have significantly higher thermal values than non-cloud pixels, indicating proper cloud detection based on temperature

### CODE REVIEW

**Review by petrasovaa — CHANGES_REQUESTED** [review]
Please also test the actual values (e.g. through assertRasterFitsUnivar)

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            output=self.output_cloud,
+            b45ratio=1.0,
+            b56composite=225,
+            overwrite=True,
+        )
+        self.runModule(
+            "r.mapcalc",
+            expression=f"{self.output_cloud} = if({self.output_cloud} > 0, 1, 0)",
+            overwrite=True,
+        )
```
Comment (petrasovaa) [inline imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:70]: What is the purpose of this?

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            output=cloud_output,
+            overwrite=True,
+        )
+        cloud_array = array.array(cloud_output)
+        self.assertGreater(
+            np.sum(cloud_array),
+            0,
+            "Cloud detection should identify at least some cloud pixels",
+        )
+        if np.sum(cloud_array) > 0 and np.sum(cloud_array) < np.size(cloud_array):
```
Comment (petrasovaa) [inline imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:191]: Why this if?

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            input=self.input_basename,
+            output=cloud_output,
+            overwrite=True,
+        )
+        cloud_array = array.array(cloud_output)
+        self.assertGreater(
+            np.sum(cloud_array),
+            0,
+            "Cloud detection should identify at least some cloud pixels",
+        )
```
Comment (petrasovaa) [inline imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:190]: This is pretty much testing the same thing as in the first test

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 2 above)
Comment (codersmiti) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:191]: The condition if np.sum(cloud_array) > 0 and np.sum(cloud_array) < np.size(cloud_array): is used to confirm that cloud detection is neither completely empty nor marking everything as a cloud. If the sum was 0, it would indicate that no clouds were detected. If the sum equals the total number of pixels, it would mean the entire area is classified as a cloud

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 1 above)
Comment (codersmiti) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:70]: This is done to follow the standard Landsat approach where 1 means clouds and 0 means clear areas. It is used to turn the cloud layer into a simple binary map. If a pixel has a value greater than 0 (meaning it’s likely a cloud), it’s set to 1. If it’s 0 (no cloud), it stays 0. It also ensures that later calculations, like those using assertRasterFitsUnivar, match the expected values—such as a maximum of 1 and a sum of 2401. Without this step, the sum could be like 10,000

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 3 above)
Comment (codersmiti) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:190]: This one specifically focuses on verifying the correlation between cloud pixels and thermal band values. The goal is to confirm that detected clouds align with expected thermal characteristics, beyond just checking if cloud pixels are detected.

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 2 above)
Comment (petrasovaa) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:191]: Right, but if the condition fails, no assert is going to be called.

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 1 above)
Comment (petrasovaa) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:70]: I see, I think you need to be more explicit in the test documentation what you are doing, 2401 looks arbitrary, it needs to be 2500, row() is indexed from 1.

Comment (codersmiti) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py:70]: Yes. Since the row() and col() are 1-indexed, the condition row() < 50 && col() < 50 implies that the rows and columns range from 1 to 49. As a result, the area covered by this condition is 49x49, which equals 2401 pixels. To clarify this, I will update the test documentation to provide a clear explanation.

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 3 above)
Comment (codersmiti) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py]: I have merged the thermal correlation check into the first test, so both the cloud mask creation and thermal validation are now handled in a single test.

File: imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py (hunk 2 above)
Comment (codersmiti) [reply imagery/i.landsat.acca/testsuite/test_i_landsat_acca.py]: I understood your point and have removed the if condition and merged the logic into the first test.

**Review by petrasovaa — APPROVED** [review]
